desktop-widgets/downloadfromdivecomputer.cpp

```diff
@@ -479,6 +481,15 @@ void DownloadFromDCWidget::updateDeviceEnabled()
 
 	// call dc_descriptor_get_transport to see if the dc_transport_t is DC_TRANSPORT_SERIAL
 	if (dc_descriptor_get_transports(descriptor) & DC_TRANSPORT_SERIAL) {
+#if defined(SERIAL_FTDI)
+		if (guessIfFTDI(descriptor)) {
+			// this should be an FTDI device that we'll connect to by using
+			// the magic device name "ftdi"
+			ui.device->setEditText("ftdi");
+			ui.device->setEnabled(false);
```

Comment:
I think this is the wrong way of doing it.

Ex, a lot of the DC_FAMILY_HW_OSTC3 devices are bluetooth, and not ftdi.

It would be better I guess to leave the field editable, so the user has the oppertunity to choose something else.

Reply:
look above in line 484 - we only do this if the descriptor includes DC_TRANSPORT_SERIAL
And for dual mode dive computers, you still can click on Bluetooth and that overrides anything we do here, anyway.
So I _**think**_ this should work as intended.